napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1620 @@
+            self.refresh()
+        self.events.face_color()
+
+    @property
+    def opacity(self):
+        """float: Opacity value between 0.0 and 1.0.
+        """
+        return self._opacity
+
+    @opacity.setter
+    def opacity(self, opacity):
+        if not 0.0 <= opacity <= 1.0:
+            raise ValueError('opacity must be between 0.0 and 1.0; '
+                             f'got {opacity}')
+
+        self._opacity = opacity
+        if self._update_properties is True:
+            index = self.selected_shapes
+            for i in index:
+                self.data.update_opacity(i, opacity)
+            self.refresh()
+        self.events.opacity()
+
+    @property
+    def selected_shapes(self):
+        """list: list of currently selected shapes
+        """
+        return self._selected_shapes
+
+    @selected_shapes.setter
+    def selected_shapes(self, selected_shapes):
+        self._selected_shapes = selected_shapes
+        self._selected_box = self.interaction_box(selected_shapes)
+
+        # Update properties based on selected shapes
+        face_colors = list(set([self.data.shapes[i]._face_color_name for i in
+                           selected_shapes]))
+        if len(face_colors) == 1:
+            face_color = face_colors[0]
+            with self.block_update_properties():
+                self.face_color = face_color
+
+        edge_colors = list(set([self.data.shapes[i]._edge_color_name for i in
+                           selected_shapes]))
+        if len(edge_colors) == 1:
+            edge_color = edge_colors[0]
+            with self.block_update_properties():
+                self.edge_color = edge_color
+
+        edge_width = list(set([self.data.shapes[i].edge_width for i in
+                          selected_shapes]))
+        if len(edge_width) == 1:
+            edge_width = edge_width[0]
+            with self.block_update_properties():
+                self.edge_width = edge_width
+
+        opacities = list(set([self.data.shapes[i].opacity for i in
+                         selected_shapes]))
+        if len(opacities) == 1:
+            opacity = opacities[0]
+            with self.block_update_properties():
+                self.opacity = opacity
+
+    @property
+    def mode(self):
+        """str: Interactive mode. The normal, default mode is PAN_ZOOM, which
+        allows for normal interactivity with the canvas. The SELECT mode allows
+        for entire shapes to be selected, moved and resized. The DIRECT mode
+        allows for shapes to be selected and their individual vertices to be
+        moved. The VERTEX_INSERT and VERTEX_REMOVE modes allow for individual
+        vertices either to be added to or removed from shapes that are already
+        selected. Note that shapes cannot be selected in this mode. The
+        ADD_RECTANGLE, ADD_ELLPISE, ADD_LINE, ADD_PATH, and ADD_POLYGON modes
+        all allow for their corresponding shape type to be added.
```

Comment:
Ideally, I would argue that these modes should go in the enum class doc, and/or the setter. Or do the docs render this as one thing? Probably I would also make some paragraphs.

Reply:
I have added paragraphs and coped the doc string to the enum class. It is now present in both